case-lib/lib.sh

```diff
@@ -810,9 +810,31 @@ is_firmware_file_zephyr()
     test "$znum" -gt 10
 }
 
+is_ipc4_debugfs()
+{
+    local ipc_type
+
+    ipc_file=/sys/kernel/debug/sof/ipc_type
+    ipc_type=$(cat $ipc_file) || {
+        return 1
+    }
+
+    if [ "$ipc_type" -eq 1 ]; then
+        return 0
+    fi
+    return 1
```

Comment:
Or keep the syntax simple for the rest of us who just understand if then...

The simplest way to write a branch is the best... 

I vote to disallow any 'smart' bash construct which requires people like me to consult an online help. bash is a means to an end.

Reply:
I must differ here sorry.

I totally agree that bash is quirky and the confusion between booleans and error codes plays a large role. It is a real pain. However, learning about that particular confusion between booleans and error codes is unfortunately not optional. especially not in sof-TEST code, the main purpose of any test code being to deal with _errors_.

For instance, until a person can make sense of what I wrote in #312, then they're not allowed to have an informed opinion on this topic! I think that boolean/error code knowledge is really the minimum required. Note I just wrote "make sense" of #312, _not_ "remember it all".

So now back to this particular example: wrapping `test_something` into `if test_something; then return 0; else return 1` is:
- "useful" only for people who MISS this basic boolean/error code knowledge.
- very verbose because you'd have to do this wrapping pretty much everywhere
- distracting because it's exposing the fact that 0 = true instead of abstracting it away.


All this being said, a one-line comment never hurts. For example:

```shell
is_something()
{
   # stuff

   # return true if X exists
   test -e X
}
```

Reply:
You're entitled to your opinions.

I personally don't know *anything* to bash and everything I read about it gives me a migraine. I won't remember any of this next Monday anyways.

so if I can use if/else I will...